### PR TITLE
implement a timer to collect PAYG data of the Uyuni host and copy them in the container

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

For PAYG we need some data from the host available in the container.
This is a timer service which collect the data and write it into a volume which is used by the container

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24209

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by admd** [comment]
This doesn't seem very extensive and appears to be pretty self-contained. I don't foresee a significant risk of introducing any regressions. I know it's late but I believe it's okay to include from my perspective.

But I would love to hear from @deneb-alpha @nkrinner @rjmateus and if they are also ok, we should go ahead and include it in GMC.

**Comment by rjmateus** [comment]
In terms of technical impact, as you say is really low. This will be needed only for PAYG scenario. Having this available on GMC would allow us to move forward on the PAYG image as soon public cloud is ready for it. 
In terms of testing, we didn't start testing it yet, since the cloud image is not yet available.
From my point of view, I think we can add it to GMC. But let ear from others :)

**Comment by mcalmer** [comment]
As this is a new package, the changelog error can be ignored as I have to add this file

File: uyuni/uyuni-payg-timer/uyuni-payg-timer.service [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Unit]
+Description=Collect PAYG data and copy them into the uyuni container
+After=uyuni-server.service csp-billing-adapter.service
+
+[Service]
+Type=oneshot
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply uyuni/uyuni-payg-timer/uyuni-payg-timer.service:6]: Most examples use oneshot or nothing.

File: uyuni/uyuni-payg-timer/uyuni-payg-extract-data.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+            modifiedPackages = modifiedPackages or has_package_modifications(
+                "python3-csp-billing-adapter-amazon"
+            )
+        elif cloudProvider == "AZURE":
+            modifiedPackages = modifiedPackages or has_package_modifications(
+                "python3-csp-billing-adapter-azure"
+            )
+        billing_service_running = is_service_running("csp-billing-adapter.service")
+
+    compliant = billing_service_running and not modifiedPackages
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply uyuni/uyuni-payg-timer/uyuni-payg-extract-data.py:225]: Good catch.  I will add that.

**Review by deneb-alpha — CHANGES_REQUESTED** [review]
Some  notes from my side:
1. I left a few code suggestion
2. there's an rpmlint warning about `suse-missing-rclink`
3. for uyuni-base, the other source under  the  same  uyuni subfolder, we  only have the base  subfolder with the  sources. for this the subfolder is called `uyuni-payg-timer`. not sure if this is intended but  I was  expecting to  see  this new source in a  subfolder called `payg-timer` not `uyuni-payg-timer`
4. For pushing from  git to IBS/OBS, we need also the package under `rel-eng/packages`. it's content should be `5.0.0 /uyuni/uyuni-payg-timer` or `5.0.0 /uyuni/payg-timer` if you will rename the  subfolder. this is needed  by `tito build`

File: uyuni/uyuni-payg-timer/uyuni-payg-timer.service (hunk 1 above)
[the remark this answers is not in the document]
Comment (mackdk) [reply uyuni/uyuni-payg-timer/uyuni-payg-timer.service:6]: I'm not a systemd expert by any means, so it was mostly a question out of curiosity. If I understand correctly, if you don't specify anything it should mean `simple`.

File: uyuni/uyuni-payg-timer/uyuni-payg-timer.spec
[the remark this answers is not in the document]
Comment (mcalmer) [reply uyuni/uyuni-payg-timer/uyuni-payg-timer.spec:1]: AFAIK this is old style and a lot of packages don't have this anymore.
And this is not a real service, just a timer. I think it is better to not have the RC link.

**Comment by mcalmer** [comment]
> Some notes from my side:
> 
> 2. there's an rpmlint warning about `suse-missing-rclink`

I checked the timer services which are installed on my system. More than the half do not have this rclink.
I think it does not make sense for timer. Only for services which really start a daemon process.

> 3. for uyuni-base, the other source under  the  same  uyuni subfolder, we  only have the base  subfolder with the  sources. for this the subfolder is called `uyuni-payg-timer`. not sure if this is intended but  I was  expecting to  see  this new source in a  subfolder called `payg-timer` not `uyuni-payg-timer`

No, I think the "base" folder was caused by python requirements which are now also obsolete. I like to have the full name as directory.

> 4. For pushing from  git to IBS/OBS, we need also the package under `rel-eng/packages`. its content should be `5.0.0 /uyuni/uyuni-payg-timer` or `5.0.0 /uyuni/payg-timer` if you will rename the  subfolder. this is needed  by `tito build`

Yes, this happens when you to the first time "tito tag".
I am not sure if I should do it or you?
But should we do this directly on master after it is merged?

**Comment by deneb-alpha** [comment]
> > Some notes from my side:
> > 
> > 2. there's an rpmlint warning about `suse-missing-rclink`
> 
> I checked the timer services which are installed on my system. More than the half do not have this rclink. I think it does not make sense for timer. Only for services which really start a daemon process.

ok, I can check with autobuild, TBH, I don't know. I only saw the warning.

> 
> > 3. for uyuni-base, the other source under  the  same  uyuni subfolder, we  only have the base  subfolder with the  sources. for this the subfolder is called `uyuni-payg-timer`. not sure if this is intended but  I was  expecting to  see  this new source in a  subfolder called `payg-timer` not `uyuni-payg-timer`
> 
> No, I think the "base" folder was caused by python requirements which are now also obsolete. I like to have the full name as directory.

sure, for me it works. I wasn't sure if we had any guideline on folder naming. but if we  don't have any "rule",  it's ok  for  me to  call it like that.

> 
> > 4. For pushing from  git to IBS/OBS, we need also the package under `rel-eng/packages`. its content should be `5.0.0 /uyuni/uyuni-payg-timer` or `5.0.0 /uyuni/payg-timer` if you will rename the  subfolder. this is needed  by `tito build`
> 
> Yes, this happens when you to the first time "tito tag". I am not sure if I should do it or you? But should we do this directly on master after it is merged?

I know  that tito  tag  creates it,  but tito build that is called for pushing the sources to IBS and OBS will  fail to push. for all  the previous new packages the PR provided also the file  under  `rel-eng/packages`. Given that you  are not going to rename the  folder, the content of `rel-eng/packages/uyuni-payg-timer` needs to be  `5.0.0 /uyuni/uyuni-payg-timer`

**Comment by deneb-alpha** [comment]
> > > Some notes from my side:
> > > 
> > > 2. there's an rpmlint warning about `suse-missing-rclink`
> > 
> > 
> > I checked the timer services which are installed on my system. More than the half do not have this rclink. I think it does not make sense for timer. Only for services which really start a daemon process.
> 
> ok, I can check with autobuild, TBH, I don't know. I only saw the warning.

Asked in packaging channel, they confirmed it  can  be  ignored  (if you can it  will be good to directly add  the rpmlintrc file, if no,  I'll do it later before next  milestone) https://suse.slack.com/archives/C02D1374S3V/p1714733983626009

File: uyuni/uyuni-payg-timer/uyuni-payg-timer.spec
[the remark this answers is not in the document]
Comment (mcalmer) [reply uyuni/uyuni-payg-timer/uyuni-payg-timer.spec:1]: I have added the rpmlintrc

**Comment by mcalmer** [comment]
As I want to keep the tag, I pushed it into uyuni repo and opened a new PR as https://github.com/uyuni-project/uyuni/pull/8691

**Pull request closed without merging**